src/self-managed-debug/src/system_catalog_dumper.rs

```diff
@@ -206,3 +217,577 @@ pub fn spawn_sql_port_forwarding_process(
+        category: RelationCategory::Introspection,
+    },
+    // Relations that are redundant with some of the above, but
+    // are used by the Console.
+    Relation {
+        name: "mz_cluster_replica_metrics_history",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_webhook_sources",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_cluster_replica_history",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_source_statistics",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_cluster_deployment_lineage",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_show_indexes",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_relations",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_frontiers",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_console_cluster_utilization_overview",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_columns",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_kafka_sources",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_kafka_sinks",
+        category: RelationCategory::Basic,
+    },
+];
+
+const TIMEOUT: Duration = Duration::from_secs(60);
+
+#[derive(Debug, Clone)]
+pub struct ClusterReplica {
+    // TODO (debug_tool1): Use this when scraping introspection relations
+    pub _cluster_name: String,
+    pub _replica_name: String,
+}
+
+pub struct SystemCatalogDumper<'n> {
+    context: &'n Context,
+    pg_client: Arc<PgClient>,
+    // TODO (debug_tool1): Use this when scraping introspection relations
+    _cluster_replicas: Vec<ClusterReplica>,
+    _pg_conn_handle: JoinHandle<Result<(), tokio_postgres::Error>>,
+}
+
+impl<'n> SystemCatalogDumper<'n> {
+    pub async fn new(context: &'n Context, connection_string: &str) -> Result<Self, anyhow::Error> {
+        let (pg_client, pg_conn) = retry::Retry::default()
+            .max_duration(TIMEOUT)
+            .retry_async_canceling(|_| async move {
+                let pg_config = &mut PgConfig::from_str(connection_string)?;
+                pg_config.connect_timeout(TIMEOUT);
+                let tls = make_tls(pg_config)?;
+                pg_config.connect(tls).await.map_err(|e| anyhow::anyhow!(e))
+            })
+            .await?;
+
+        info!("Connected to PostgreSQL server at {}...", connection_string);
+
+        let handle = task::spawn(|| "postgres-connection", pg_conn);
+
+        // Set search path to system catalog tables
+        pg_client
+            .execute(
+                "SET search_path = mz_internal, mz_catalog, mz_introspection",
+                &[],
+            )
+            .await?;
```

Comment:
would be nice to add some `.context("set search path")` here, and to the other DB calls! Otherwise when we get an error back we don't know which query caused the error to fail